RFM9x.nut

```diff
@@ -0,0 +1,381 @@
+    const RFM9X_FIFO = 0x00;
+
+    // SPI masks
+    const RFM9X_WRITE_MASK = 0x80;
+    const RFM9X_READ_MASK = 0x00;
+    
+    const RFM9X_REG_OP_MODE = 0x01;
+    // f_RF = ((X_OSC) * Fr_f ) / (2^19). Resolution is 61.035 Hz if X_OSC = 32MHz
+    const RFM9X_REG_FREQUENCY_HIGH = 0x06; // MSB (3 bytes long, contiguous)
+    const RFM9X_REG_PA_CONFIG = 0x09;
+    const RFM9X_REG_PA_RAMP = 0x0A;
+    const RFM9X_REG_OCP = 0x0B;
+    const RFM9X_REG_LNA = 0x0C;
+    const RFM9X_REG_FIFO_ADDR_PTR = 0x0D;
+    const RFM9X_REG_FIFO_TX_BASE_ADDR = 0x0E; // write base address in RFM9X_FIFO data buffer
+    const RFM9X_REG_FIFO_RX_BASE_ADDR = 0x0F; // read base address in RFM9X_FIFO data buffer
+    const RFM9X_REG_FIFO_RX_CURRENT_ADDR = 0x10; // Start address of last packet received
+    
+    const RFM9X_REG_IRQ_FLAGS_MASK = 0x11;
+    // IRQ flags
+    const RFM9X_REG_IRQ_FLAGS = 0x12;
+    
+    // Number of bytes in most recent payload
+    const RFM9X_REG_RX_N_BYTES = 0x13; 
+    
+    const RFM9X_REG_RX_HEADER_CNT_VALUE_MSB = 0x14;
+    const RFM9X_REG_RX_HEADER_CNT_VALUE_LSB = 0x15;
+    const RFM9X_REG_RX_PACKET_CNT_VALUE_MSB = 0x16;
+    const RFM9X_REG_RX_PACKET_CNT_VALUE_LSB = 0x17;
+    
+    const RFM9X_REG_MODEL_STAT = 0x18;
+    const RFM9X_REG_PKT_SNR_VALUE = 0x19;
+    const RFM9X_REG_PKT_RSSI_VALUE = 0x1A;
+    const RFM9X_REG_RSSI_VALUE = 0x1B;
+    const RFM9X_REG_HOP_CHANNEL = 0x1C;
+    const RFM9X_REG_MODEM_CONFIG1 = 0x1D;
+    const RFM9X_REG_MODEM_CONFIG2 = 0x1E;
+    const RFM9X_REG_MODEM_CONFIG3 = 0x26;
+    const RFM9X_REG_SYMB_TIMEOUT_LSB = 0x1F;
+    
+    const RFM9X_REG_PREAMBLE_MSB = 0x20;
+    const RFM9X_REG_PREAMBLE_LSB = 0x21;
+    const RFM9X_REG_PAYLOAD_LENGTH = 0x22;
+    const RFM9X_REG_MAX_PAYLOAD_LENGTH = 0x23;
+    const RFM9X_REG_HOP_PERIOD = 0x24;
+    const RFM9X_REG_FIFO_RX_BYTE_ADDR = 0x25;
+    
+    
+    // -------------------- Shared Registers ---------------------- //
+
+    // DIO mappings are used for interrupts
+
+    const RFM9X_REG_DIO_MAPPING1 = 0x40;
+    const RFM9X_REG_DIO_MAPPING2 = 0x41;
+    const RFM9X_REG_VERSION = 0x42;
+    const RFM9X_REG_TCXO = 0x4B;
+    
+    // Default init constants
+    const RFM9X_DEFAULT_FREQUENCY = 915000000;
+    const RFM9X_DEFAULT_BANDWIDTH = "125";
+    const RFM9X_DEFAULT_SF = 7;
+    const RFM9X_DEFAULT_CODING_RATE = "4/5";
+    const RFM9X_DEFAULT_IMPLICIT_HEADER_MODE = false;
+    const RFM9X_DEFAULT_PREAMBLE_LENGTH = 8;
+
+class RFM9x {
+    
+    _spiModule = null;
+    _cs = null;
+    _irqPin = null;
+    _receiveHandler = null;
+    _sendFinished = true;
+    _error = false;
+
+    static FREQ_STEP = 61.035;
```

Comment:
These static variables could all be constants or enums. 
https://electricimp.com/docs/squirrel/squirrelcrib/